scripts/resetdb.sh

```diff
@@ -34,7 +34,7 @@ collect_vars() {
   [[ ${VERBOSE} = 'true' ]] && echo "- Using MySQL Flags: ${FLAGS[@]}"
 
   # append password if supplied
-  [ -z ${DB_PASSWORD+x} ] || FLAGS+=("-p'${DB_PASSWORD}'")
+  [ -z ${DB_PASSWORD+x} ] || FLAGS+=(-p"${DB_PASSWORD}")
```

Comment:
I don't think this will work if `$DB_PASSWORD` contains a space.

Reply:
Having said that, the current way of doing it won't work if `$DB_PASSWORD` contains an apostraphe...

Reply:
It looks like the correct way to do this is with `${DB_PASSWORD@Q}` (see the bottom of https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html), except that was only added in Bash 4.4 and we're using Bash 4.3 :(

Reply:
The thing is, we don't need quotes in this place because below in "${FLAGS[@]}" the parameters get quoted again (which is the reason why the previous version of the script didn't work even for non space-containing passwords). This way, mysql gets '-pMY PASSWORD WITH SPACES' as a single parameter and successfully parses it.

Moreover, yesterday Gary and I observed that the new script works with space-containing passwords.

Reply:
Ah ok, I'd missed that it re-quotes everything further down. LGTM.